### ISSUE TITLE
ValueError

### ISSUE THREAD

**Issue by usmanali45** [nikochiko/code#1]
![img1](https://user-images.githubusercontent.com/51873788/106048833-306c4780-6107-11eb-9b4b-bfd4a3cf7c81.PNG)
![img2](https://user-images.githubusercontent.com/51873788/106048862-38c48280-6107-11eb-900e-b0adf45f6944.PNG)

Hi! This is my error log ValueError comes on epoch 1. I am doing it on google colab. Tried many solutions from internet but nothing worked. Can you figure out what's wrong here?

Thank you!

**Comment by nikochiko** [comment]
Hey @usmanali45, sure I'll be glad to help. Can you let me know which notebook you are trying to run?

P.S. If you got here from a thread in tensorflow/tensorflow, can you also link to that so I can get some context.

**Comment by usmanali45** [comment]
Here is the link of the notebook file which i am trying to run:
https://github.com/MITESHPUTHRANNEU/Speech-Emotion-Analyzer/blob/master/final_results_gender_test.ipynb

I actually closed all the tabs and don't remember which thread i've got her through

**Comment by usmanali45** [comment]
I got it. Here's the thread which i used to get here, 

https://github.com/tensorflow/tensorflow/issues/34003

**Comment by nikochiko** [comment]
Hey @usmanali45 thanks for finding the links! 

I think I'm pretty sure the error is because the X_train and y_train got incoherent shapes somewhere. If you look at the run numbers on the notebook you shared, that particular cell causing the error hasn't been run at all. The rest of the notebook should probably run fine without that cell. 

That being said, if you do want to figure out what caused the error, quite sure it should be somewhere in the code logic.

Thank you!

**Comment by nikochiko** [comment]
Yes. The X_traincnn has only been defined with expanded dimensions in the cell just above that cell, so that might have been the issue.

**Comment by usmanali45** [comment]
@nikochiko Thank you very much for responding. You were right x and y got incoherent shapes. But problem is still not solved. I've checked the shapes of x and y they are different from each other. Here is the code snippet,

![img1](https://user-images.githubusercontent.com/51873788/106207345-6da60780-61e3-11eb-8909-d00a99dab961.PNG)

I dont understand why they've got incoherent shapes. I guess it is because I've converted y_train and y_test to .astype(str)
But if I remove .astype(str) another error comes.  Can you please figure out what can be wrong here. Here's the screenshot of it,

![img2](https://user-images.githubusercontent.com/51873788/106207848-3dab3400-61e4-11eb-8953-693991502aea.PNG)
![img3](https://user-images.githubusercontent.com/51873788/106207877-47cd3280-61e4-11eb-99fa-6bd928a0fa4d.PNG)
